CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -383,6 +383,13 @@ public function postProcess() {
     }
     else {
       $result = CRM_Contribute_BAO_Contribution::recordAdditionalPayment($this->_contributionId, $submittedValues, $this->_paymentType, $participantId);
+      // Fetch the contribution & do proportional line item assignment
+      $params = array('id' => $this->_contributionId);
+      $contribution = CRM_Contribute_BAO_Contribution::retrieve($params, CRM_Core_DAO::$_nullArray, $params);
```

Comment:
It's not a blocker to merge - but avoid that $_nullArray if possible. It is passed to a bunch of functions that require references in core and some of them actually add values to it - which hang around. It can lead to really hard to track down errors. There is a ticket somewhere about getting rid of it.